### PR TITLE
Variant metrics prod ready

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

* added tests for metrics
* bump up the version since this change is already tested and ready to go live
* stopped fake unleash from initializing parent unleash, so we don't need to disable metrics and all open handles errors from jest are gone

This PR builds upon https://github.com/Unleash/unleash-proxy/pull/131 that made the proxy more testable

### CODE REVIEW

File: src/test/client.test.ts [hunk 1]
```diff
@@ -180,3 +180,40 @@ test('should return default variant for disabled toggles', () => {
...
+        ['toggle', true],
+        ['toggle', true],
+        ['toggle', true],
+        ['toggle', false],
+    ]);
+    expect(metrics.recordedCountVariant).toStrictEqual([
+        ['toggle', 'variantA'],
+        ['toggle', 'variantA'],
+        ['toggle', 'variantB'],
+        ['toggle', 'disabled'],
```
Comment (thomasheartman) [inline src/test/client.test.ts:217]: Is the order significant here? Would it be better to compare counts? Is the storage the same in the FakeMetrics class as in the real Metrics class?

**Review by thomasheartman — APPROVED** [review]
Looks pretty safe to me! Just a small question regarding the testing, but no blockers 🙌🏼

File: src/test/client.test.ts (hunk 1 above)
Comment (kwasniew) [reply src/test/client.test.ts:217]: FakeMetrics is a very stupid mock recording all invocations. Another alternative would be to have a smarter mock summing up the numbers. I decided to go for the simples fake replacement possible in this case. If you feel strongly about recording the actual numbers I may reconsider my tradeoff.

Comment (thomasheartman) [reply src/test/client.test.ts:217]: Got it. I don't really feel strongly one way or the other, so no need to change it for me. Just curious 😄

File: src/test/unleash.mock.ts [hunk 2]
```diff
@@ -15,6 +15,7 @@ class FakeUnleash extends Unleash {
     // fix constructor
     constructor(unleashConfig: UnleashConfig) {
         super(unleashConfig);
+        super.destroy(); // prevent parent constructor initialization
```
Comment (kwasniew) [inline src/test/unleash.mock.ts:18]: this line is important as Unleash parent fetches toggles and starts metrics reporting that we don't want

**Review by ivarconr — APPROVED** [review]

**Pull request merged**